print-radius.c

```diff
@@ -671,6 +673,15 @@ print_attr_string(netdissect_options *ndo,
            data++;
            length--;
         break;
+      case EAP_MESSAGE:
+           ND_TCHECK_2(data + 2);
```

Comment:
This code block is wrong because it does not check `length` before accessing into `data`, and because it likely does not need to access into `data` in the first place as I tried to explain in August. Also this does not seem to be a good place for this block because when the EAP data is shorter than the RADIUS attribute, the rest of the attribute will be printed as a string (unless I misunderstood the code flow, feel free to correct me); it seems more appropriate for `print_attr_strange()`.

Reply:
Thanks for the review again. Yes, agree with your point. 

Made code changes to avoid accessing the data part outside of eap_print() API (for maintained layered approach). 

Also, in cases where the EAP data is lesser than RADIUS attribute, its an erroneous scenario and nd_print_trunc() will be called by eap_print() API itself and hence, as such, no more printing will be required. 

Kindly review the changes and let me know your views.

Thanks a lot for the continued support.

Reply:
Hi @infrastation 

Kindly review the changes and let me know your views.

Reply:
Hi @infrastation @guyharris 

Kindly review the changes and let me know your views.